src/Http/Http/src/Features/FormFeature.cs

```diff
@@ -71,7 +70,21 @@ internal FormFeature(HttpRequest request, FormOptions options, Endpoint? endpoin
     {
         get
         {
-            _ = MediaTypeHeaderValue.TryParse(_request.ContentType, out var mt);
+            MediaTypeHeaderValue? mt = default;
+
+            // Set directly
+            if (Form is not null)
+            {
+                mt = new MediaTypeHeaderValue("application/x-www-form-urlencoded");
+            }
```

Comment:
Does this happen multiple times if _request is null?

Reply:
@davidfowl, thanks for spotting that. I have changed the logic. I also felt that the sequence of things should be different in this getter. If _request is not null, it's content type should take precedence, in my view. But there may be reasons I am not aware of for not changing this.

In general, I feel the fact that there is a public setter for the Form property makes the design of this type awkward, but I guess it might be needed? Because of that, you could actually end up with an instance of this type where both _form and _request is null.  Should the setter for example throw if you try to set Form to null, like the constructor does?